src/Bridge/Graphql/Resolver/ItemResolverFactory.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Graphql\Resolver;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use GraphQL\Type\Definition\ResolveInfo;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Creates a function retrieving an item to resolve a GraphQL query.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ *
+ * @internal
+ */
+final class ItemResolverFactory extends AbstractResolverFactory implements ItemResolverFactoryInterface
+{
+    private $itemDataProvider;
+    private $normalizer;
+    private $identifiersExtractor;
+
+    public function __construct(ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider, NormalizerInterface $normalizer, IdentifiersExtractorInterface $identifiersExtractor)
+    {
+        parent::__construct($subresourceDataProvider);
+
+        $this->itemDataProvider = $itemDataProvider;
+        $this->normalizer = $normalizer;
+        $this->identifiersExtractor = $identifiersExtractor;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function createItemResolver(string $resourceClass, string $rootClass): callable
+    {
+        return function ($root, $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass) {
+            $rootProperty = $info->fieldName;
+            $rootIdentifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($rootClass);
+            if (isset($root[$rootProperty])) {
+                $item = $this->getSubresource($rootClass, $root, $rootIdentifiers, $rootProperty, $resourceClass, false);
+            } else {
+                $identifiers = [];
+                $uniqueIdentifier = [];
+                foreach ($rootIdentifiers as $rootIdentifier) {
+                    if (!isset($args[$rootIdentifier])) {
+                        continue;
+                    }
+
+                    if (\is_array($args[$rootIdentifier])) {
+                        if (\count($args[$rootIdentifier]) > 1) {
```

Comment:
Same on these, we currently are not prefixing with `\` in the base code.

Reply:
So we should add and enable this fixer in our php-cs-fixer config!

Reply:
Yes but it is a risky one, IIRC, it breaks our tests.

Reply:
wdyt we remove `\` for now and we will do this in another pr.

Reply:
Agreed!